SofaKernel/framework/sofa/simulation/Node.h

```diff
@@ -551,13 +551,35 @@ class SOFA_SIMULATION_CORE_API Node : public sofa::core::objectmodel::BaseNode,
 
     std::stack<Visitor*> actionStack;
 
-    virtual void notifyAddChild(Node::SPtr node);
-    virtual void notifyRemoveChild(Node::SPtr node);
-    virtual void notifyMoveChild(Node::SPtr node, Node* prev);
-    virtual void notifyAddObject(sofa::core::objectmodel::BaseObject::SPtr obj);
-    virtual void notifyRemoveObject(sofa::core::objectmodel::BaseObject::SPtr obj);
-    virtual void notifyMoveObject(sofa::core::objectmodel::BaseObject::SPtr obj, Node* prev);
-    virtual void notifySleepChanged();
+    virtual void notifyAddChild(Node::SPtr parent, Node::SPtr child) final ;
+    virtual void notifyRemoveChild(Node::SPtr parent, Node::SPtr child) final;
+    virtual void notifyMoveChild(Node::SPtr parent, Node::SPtr child, Node* prev_parent) final;
+    virtual void notifyAddObject(Node::SPtr parent, sofa::core::objectmodel::BaseObject::SPtr obj) final;
+    virtual void notifyRemoveObject(Node::SPtr parent, sofa::core::objectmodel::BaseObject::SPtr obj) final;
+    virtual void notifyMoveObject(Node::SPtr parent, sofa::core::objectmodel::BaseObject::SPtr obj, Node* prev_parent) final;
+    virtual void notifySleepChanged(Node* node) final;
+
+    virtual void doNotifyAddChild(Node::SPtr parent, Node::SPtr child);
+    virtual void doNotifyRemoveChild(Node::SPtr parent, Node::SPtr child);
+    virtual void doNotifyMoveChild(Node::SPtr parent, Node::SPtr child, Node* prev_parent);
+    virtual void doNotifyAddObject(Node::SPtr parent, sofa::core::objectmodel::BaseObject::SPtr obj);
+    virtual void doNotifyRemoveObject(Node::SPtr parent, sofa::core::objectmodel::BaseObject::SPtr obj);
+    virtual void doNotifyMoveObject(Node::SPtr parent, sofa::core::objectmodel::BaseObject::SPtr obj, Node* prev_parent);
+
+    virtual void notifyAddChildDone(Node::SPtr parent, Node::SPtr child);
+    virtual void notifyRemoveChildDone(Node::SPtr parent, Node::SPtr child);
+    virtual void notifyMoveChildDone(Node::SPtr parent, Node::SPtr child, Node* prev_parent);
+    virtual void notifyAddObjectDone(Node::SPtr parent, sofa::core::objectmodel::BaseObject::SPtr obj);
+    virtual void notifyRemoveObjectDone(Node::SPtr parent, sofa::core::objectmodel::BaseObject::SPtr obj);
+    virtual void notifyMoveObjectDone(Node::SPtr parent, sofa::core::objectmodel::BaseObject::SPtr obj, Node* prev_parent);
+
+//    virtual void doNotifyAddChildDone(Node::SPtr parent, Node::SPtr child);
```

Comment:
Since there's no implementation of the notifyAddChildDone method, those delegates are useless.. although in terms of code symmetry, it's kind of wierd not to have it..